mne_bids/tests/test_path.py

```diff
@@ -166,6 +168,118 @@ def test_get_entity_vals(entity, expected_vals, kwargs, return_bids_test_dir):
     shutil.rmtree(deriv_path)
 
 
+def test_path_benchmark(tmp_path_factory):
+    """Benchmark exploring bids tree."""
+    tmp_bids_root = tmp_path_factory.mktemp("abc")
+
+    derivatives = [Path("derivatives", "derivatives" + str(i)) for i in range(17)]
+
+    bids_subdirectories = ["", "sourcedata", *derivatives]
+
+    def equal_length_subj_ids(v):
+        assert all(v > 0)
+        n_digits = len(str(np.max(v))) + 1
+        return list(map(lambda i: "0" * (n_digits - len(str(i))) + str(i), v))
```

Comment:
I think this is not really necessary for the sake of the test, just hardcode it, I would say. You are the one creating the mock dataset below, so you know exactly how many subjects you feed into it. Apart from that, the prefixing with zeros is not strictly required :)